### PR TITLE
simplify `make manpages`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**\- What I did**
Simplified `make manpages`.

The `generator.go` hadn't used the top-level vendor directory and had managed the dependency using its own glide.lock.
Considering that now `generator.go` depends on `github.com/docker/docker/client` rather than vendored `github.com/docker/engine-api/client`, I think we should unify the vendoring system.

This PR also updates md2man to v1.0.6, which improves the rendering of hyperlinks.
https://github.com/cpuguy83/go-md2man/pull/21

**\- How I did it**
- Removed `man/Dockerfile.$GOARCH` and some scripts which were very complicated
- Moved `man/Dockerfile` to the top level directory. Now it uses the top level directory as the build context (because `generator.go` depends on `github.com/docker/docker/client`)
- Removed `man/glide.*`, and vendored packages using `hack/vendor.sh`
- Updated `generator.go` so as to make it much more pretty

**\- How to verify it**

```
$ make manpages
$ ls man/man*
man/man1:
docker-attach.1   docker-load.1                docker-node-ps.1      docker-search.1            docker-swarm-leave.1
docker-build.1    docker-login.1               docker-node-rm.1      docker-service-create.1    docker-swarm-update.1
docker-commit.1   docker-logout.1              docker-node-update.1  docker-service-inspect.1   docker-swarm.1
docker-cp.1       docker-logs.1                docker-node.1         docker-service-ls.1        docker-tag.1
docker-create.1   docker-network-connect.1     docker-pause.1        docker-service-ps.1        docker-top.1
docker-diff.1     docker-network-create.1      docker-port.1         docker-service-rm.1        docker-unpause.1
docker-events.1   docker-network-disconnect.1  docker-ps.1           docker-service-scale.1     docker-update.1
docker-exec.1     docker-network-inspect.1     docker-pull.1         docker-service-update.1    docker-version.1
docker-export.1   docker-network-ls.1          docker-push.1         docker-service.1           docker-volume-create.1
docker-history.1  docker-network-rm.1          docker-rename.1       docker-start.1             docker-volume-inspect.1
docker-images.1   docker-network.1             docker-restart.1      docker-stats.1             docker-volume-ls.1
docker-import.1   docker-node-demote.1         docker-rm.1           docker-stop.1              docker-volume-rm.1
docker-info.1     docker-node-inspect.1        docker-rmi.1          docker-swarm-init.1        docker-volume.1
docker-inspect.1  docker-node-ls.1             docker-run.1          docker-swarm-join-token.1  docker-wait.1
docker-kill.1     docker-node-promote.1        docker-save.1         docker-swarm-join.1        docker.1

man/man5:
Dockerfile.5  docker-config-json.5

man/man8:
dockerd.8
```

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

File: Dockerfile.man [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.7
```
Comment (AkihiroSuda) [inline Dockerfile.man]: TODO: add proper Dockerfiles for other archs

File: man/generate.go [hunk 2]
```diff
@@ -26,18 +35,74 @@ func generateManPages(path string) error {
...
 	})
 }
 
+func generateLegacyManPages(outPath, inPath string) error {
+	mdPaths, err := filepath.Glob(filepath.Join(inPath, "*.md"))
+	if err != nil {
+		return err
+	}
+	for _, mdPath := range mdPaths {
+		fmt.Printf("Processing %s\n", mdPath)
```
Comment (AkihiroSuda) [inline man/generate.go]: TODO: remove this printf or at least move (because README.md is skipped)

**Comment by justincormack** [comment]
cc @dnephin

**Comment by dnephin** [comment]
I'm -1 on this change. This is absolutely the opposite direction I think we should be going.

* there is no reason to put these `Dockerfile` at the top level of the repo, making them slower to build (because you have to upload a much larger context). They belong under `man/`
* there is no reason to add these "generate manpage" dependencies to the set of dependencies used to build the binary. These are different sets of dependencies. Some tools (like glide) can manage this, but our vendor.sh can not. 


Upgrading the dependency is fine, but I don't think we should make the other changes.

**Comment by AkihiroSuda** [comment]
OK, thank you for the feedback

**Pull request closed without merging**